File: pyutil/adchpp.py
import sys,os
sys.path.append(os.path.abspath('../build/debug-default/bin/'))
sys.path.append(os.path.abspath('../build/debug-mingw/bin/'))

import pyadchpp as a

from Helpers import *
from Hub import *

hub = Hub()
op = Profile('op', hub)
vip = Profile('vip', hub)

profiles = dict([(x.name, x) for x in (op, vip)])

users = [User(profile=op, nick='arnetheduck', password='test')]

nicks = dict([(x.nick, x) for x in users if x.nick])
cids = dict([(x.cid, x) for x in users if x.cid])

configPath = os.path.abspath('../etc/') + os.sep
core = a.Core.create(configPath)

def findUser(nick, cid):
	print "looking for", nick, cid
	if nick in nicks:
		return nicks[nick].password
	if cid in cids:
		return cids[cid].password
	
	return None

def handler(signum, frame):
    core.shutdown()

def run():
    print "Starting"
    sil = a.TServerInfoList(1)
    si = a.ServerInfo.create()
    si.port = 2780
    sil[0] = si
    core.getSocketManager().setServers(sil)
    pw = PasswordHandler(core, findUser, None, None)
    iv = InfVerifier(core, None, None)
    print "."
    import signal
    signal.signal(signal.SIGINT, handler)

    try:
        import plugins
        plugins.init(core)
        core.run()
    except Exception as e:
        print "\n\nFATAL: Can't start ADCH++: %s\n" % e

if __name__ == '__main__':
    run()

core = None

File: rbutil/adchpp.rb

$:.unshift(File.expand_path('../build/debug-default/bin/'))
$:.unshift(File.expand_path('../build/release-default/bin/'))

puts "Loading module..."

require 'rbadchpp'

a = Rbadchpp

config_path = File.expand_path("../etc/") + '/'
puts "Configuration path is #{config_path}..."

core = a::Core.create(config_path)

sil = a::TServerInfoList.new
si = a::ServerInfo.create()
si.port= 2780
sil.push(si)

core.get_socket_manager().set_servers(sil)
cm = core.get_client_manager()

print "."
counter = 0
connected = cm.signal_connected().connect(Proc.new { counter = counter + 1 })
disconnected = cm.signal_disconnected().connect(Proc.new { counter = counter - 1 })
cb = core.add_timed_job(1000, Proc.new { puts counter })

trap("INT") { core.shutdown() }
core.run()

puts "Shutting down..."

disconnected = nil
connected = nil
puts cb
cb.call

cb = nil

core = nil
